### ISSUE TITLE
Can't show scalar

### ISSUE THREAD

**Issue by zhangguangzhi** [PaddlePaddle/VisualDL#320]
My program uses main() to call the trained function train(), which I have in my train function. Here is my program file. When I executed, there was no expected result, Can't show scalar.

![1](https://user-images.githubusercontent.com/29686413/37380733-fa287e92-2774-11e8-982b-6438810e924a.jpg)
![2](https://user-images.githubusercontent.com/29686413/37380734-fa6462f4-2774-11e8-8ef5-6089bbde81c7.jpg)
![image](https://user-images.githubusercontent.com/29686413/37380757-0b592b4e-2775-11e8-8114-9254b1b7d2ac.png)

**Comment by zhangguangzhi** [comment]
Chinese：我的程序是使用main()来调用训练的函数train()，我在我的train函数，以下是我的程序文件。当我执行时，没有出现预想的结果，图形可显示step，但是无法显示值。
而且只能查看到最后一个step

**Comment by jetfuel** [comment]
@zhangguangzhi It seems that the t_loss was not updated. You can try to generate some fake logs manually to check if the visualDL is working
```
import random
from visualdl import ROOT, LogWriter

logdir = './scratch_log'
logw = LogWriter(logdir, 2000)

# create scalars in mode train and test.
with logw.mode('train') as logger:
    scalar0 = logger.scalar("scratch/scalar")

with logw.mode('test') as logger:
    scalar1 = logger.scalar("scratch/scalar")

# add fake scalar records. 
last_record0 = 0.
last_record1 = 0.
for step in range(1, 1000):
    last_record0 += 0.1 * (random.random() - 0.3)
    last_record1 += 0.1 * (random.random() - 0.7)
    scalar0.add_record(step, last_record0)
    scalar1.add_record(step, last_record1)

```

**Comment by yeyupiaoling** [comment]
@zhangguangzhi 
My program happen this problem just now. You can try again. close VisualDL, and then start again.

**Comment by zhangguangzhi** [comment]
@jetfuel  Virsualdl is normal
![image](https://user-images.githubusercontent.com/29686413/37385002-df68f60a-278c-11e8-8875-73fe4d0a13f0.png)

**Comment by zhangguangzhi** [comment]
@jetfuel 
I modified my code, trained five times, showed the first t_loss for the first time, and finally showed the fifth t_loss. Not a line.
![image](https://user-images.githubusercontent.com/29686413/37385274-5859329a-278e-11e8-9563-8220605b48c8.png)
first train
![image](https://user-images.githubusercontent.com/29686413/37385401-bd960b10-278e-11e8-9dff-9c5860cca404.png)
second train
![image](https://user-images.githubusercontent.com/29686413/37385479-f37fa9c0-278e-11e8-9d6b-27a58a064aa2.png)

**Comment by zhangguangzhi** [comment]
@jetfuel Thank you
I will take out this configuration and it will do.
`logger = LogWriter("./log", sync_cycle=1)
with logger.mode("t_loss") as logger:
    train_loss = logger.scalar("loss/val")`

**Comment by jetfuel** [comment]
@zhangguangzhi Ah, I see that you create the logger within the for-loop. To use VisualDL, you only need to create the scalar log writer once. Then use that scalar log writer in your training process. 

Also, it is recommended to set your sync_cycle larger if you want to record more data points. Saving to file system is expansive. The team is aware of this and is working to improve the performance.
